### PR TITLE
Updating Consul Glossary with more industry standard terms

### PR DESCRIPTION
**Pull request by jsosulska**
This is an updated rebase of #9348.

### CODE REVIEW

**Comment by watsonian** [comment]
@jsosulska A lot of this doesn't seem to be Consul-specific at all. Do we really want the Consul glossary filled with general industry definitions that don't necessarily have anything to do with Consul directly?

**Comment by swarnap** [comment]
> @jsosulska A lot of this doesn't seem to be Consul-specific at all. Do we really want the Consul glossary filled with general industry definitions that don't necessarily have anything to do with Consul directly?

Hi @watsonian - the purpose of "glossary" is to help folks understand the broad scope of technology terms related to the project/product.  This is to help folks find consul.io when they search for generic terms such as the ones listed there.  Consul-specific terms can be listed/collected/grouped under Vocabulary (the way Nomad has it here: https://learn.hashicorp.com/tutorials/nomad/get-started-vocab?in=nomad/get-started) which would be product-specific terminology.

Does this help?

**Review by i0rek — APPROVED** [review]
LGTM

File: website/content/docs/install/glossary.mdx [hunk 1]
```diff
@@ -73,3 +73,306 @@ over the internet or wide area network.
...
+developing and launching the application. 
+
+## Reverse Proxy	
+A reverse proxy handles requests coming from outside, to the internal 
+network. Reverse Proxy provides a level of security that prevents the 
+external clients from having direct access to data on the corporate servers. 
+The reverse proxy is usually placed between the web server and the external 
+traffic.  
+
+## Role-based Access Controls	"The act of restricting or provisioning access
```
Comment (i0rek) [inline website/content/docs/install/glossary.mdx:317]:
```suggestion
## Role-based Access Controls	
The act of restricting or provisioning access 
```

**Review by i0rek — APPROVED** [review]
LGTM, found a small typo

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/373743.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit bb07d1a30f95f9c9e2ba5cf3e913a6e72e63b25e onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit bb07d1a30f95f9c9e2ba5cf3e913a6e72e63b25e onto `release/1.10.x` succeeded!

**Pull request merged**